packages/cli/src/steps/compile/runWebpackCompiler.ts

```diff
@@ -82,12 +82,17 @@ function getFilePath(resource = '') {
   return path.relative(process.cwd(), resource);
 }
 
-function addDependencies(dependencyPackages) {
+function addDependencies(dependencyPackages, existingDependencies: string[] = []) {
   return (module): Module => {
     const isExternal = !!module.external;
     const isNodeModule = checkNodeModule(module.resource);
     const packageName = isNodeModule || isExternal ? getPackageName(module) : undefined;
     const filePath = module.resource && getFilePath(module.resource);
+    const isCircularImport = !!existingDependencies.find(
```

Comment:
I would have used a hash map here as lookup would be much faster than finding sth in an array

Reply:
Thought about that as well. Let's use a map :)

Reply:
I used a `Set`, as we only need the path and it also has a `has()` function :)